src/components/AppHeader/MobileMenu/index.tsx

```diff
@@ -1,25 +1,68 @@
 import { useState } from 'react';
-import { LegacyMenuHamburger2pxIcon } from '@deriv/quill-icons';
-import { Drawer, useDevice } from '@deriv-com/ui';
+import NetworkStatus from '@/components/AppFooter/NetworkStatus';
+import ServerTime from '@/components/AppFooter/ServerTime';
+import { LANGUAGES } from '@/constants';
+import { useModalManager } from '@/hooks';
+import { useTranslations } from '@deriv-com/translations';
+import { Drawer, MobileLanguagesDrawer, useDevice } from '@deriv-com/ui';
+import { BackButton } from './BackButton';
+import { MenuContent } from './MenuContent';
+import { MenuHeader } from './MenuHeader';
+import { ToggleButton } from './ToggleButton';
+
+// TODO the design inside LanguageSwitcher does not match the production => change from ui side
+// TODO fix platformswitcher issue in the tablet
+// TODO logout onclick
+// TODO adding language to the sidemenu>accountsettings
+// TODO add disable condition to the accountsettings menu
 
 export const MobileMenu = () => {
     const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+    const { currentLang = 'EN', localize, switchLanguage } = useTranslations();
+    const { hideModal, isModalOpenFor, showModal } = useModalManager();
     const { isDesktop } = useDevice();
 
+    const openDrawer = () => setIsDrawerOpen(true);
+    const closeDrawer = () => setIsDrawerOpen(false);
+
+    const openLanguageSetting = () => showModal('MobileLanguagesDrawer');
+    const isLanguageSettingVisible = Boolean(isModalOpenFor('MobileLanguagesDrawer'));
+
     if (isDesktop) return null;
     return (
         <>
-            <div className='flex items-center justify-center py-2 px-4 h-full' onClick={() => setIsDrawerOpen(true)}>
-                <LegacyMenuHamburger2pxIcon iconSize='xs' />
-            </div>
-            <Drawer
-                isOpen={isDrawerOpen}
-                onCloseDrawer={() => {
-                    setIsDrawerOpen(false);
-                }}
-                width='300px'
-            >
-                aaa
+            <ToggleButton onClick={openDrawer} />
+
+            <Drawer isOpen={isDrawerOpen} onCloseDrawer={closeDrawer} width='29.5rem'>
+                <Drawer.Header onCloseDrawer={closeDrawer}>
+                    <MenuHeader
+                        hideLanguageSetting={isLanguageSettingVisible}
+                        openLanguageSetting={openLanguageSetting}
+                    />
+                </Drawer.Header>
+
+                <Drawer.Content>
+                    {isLanguageSettingVisible ? (
+                        <>
+                            <BackButton buttonText={localize('Language')} onClick={hideModal} />
+
+                            <MobileLanguagesDrawer
+                                isOpen
+                                languages={LANGUAGES}
+                                onClose={hideModal}
+                                onLanguageSwitch={code => switchLanguage(code)}
```

Comment:
```suggestion
                                onLanguageSwitch={switchLanguage}
```

Reply:
Same as this https://github.com/deriv-com/p2p/pull/125#discussion_r1637497002

Reply:
Thanks @nada-deriv will change this 🙏